Upstream-Status: Pending

Fix compile errors when openssl disable weak encryptions support.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
diff --git a/rpmio/rpmssl.c b/rpmio/rpmssl.c
index 66e1666..359abb0 100644
--- a/rpmio/rpmssl.c
+++ b/rpmio/rpmssl.c
@@ -627,9 +627,11 @@ exit:
     if (rc == 1) {
 	/* XXX OpenSSL BUG(iirc): ensure that the asn1 flag is set. */
 	if (EVP_PKEY_type(ssl->pkey->type) == EVP_PKEY_EC) {
+#ifndef OPENSSL_NO_ECDSA
 	    EC_KEY * ec = EVP_PKEY_get1_EC_KEY(ssl->pkey);
 	    EC_KEY_set_asn1_flag(ec, OPENSSL_EC_NAMED_CURVE);
 	    EC_KEY_free(ec);
+#endif
 	}
     } else {
 	if (ssl->pkey)
@@ -734,10 +736,12 @@ assert(xx);
 	ssl->dsa = NULL;
 	break;
     case 50:		/* ECDSA r */
+#ifndef OPENSSL_NO_ECDSA
 assert(ssl->ecdsasig == NULL);
 	ssl->qbits = mbits;
 	ssl->ecdsasig = ECDSA_SIG_new();
 	ssl->ecdsasig->r = BN_bin2bn(p+2, nb-2, ssl->ecdsasig->r);
+#endif
 	break;
     case 51:		/* ECDSA s */
 #if !defined(OPENSSL_NO_ECDSA)
@@ -863,6 +867,7 @@ void rpmsslClean(void * impl)
 	    BN_free(ssl->hm);
 	ssl->hm = NULL;
 
+#ifndef OPENSSL_NO_ECDSA
 	ssl->curveN = _free(ssl->curveN);
 	ssl->nid = 0;
 	if (ssl->ecdsasig)
@@ -871,6 +876,7 @@ void rpmsslClean(void * impl)
 	if (ssl->priv)
 	    BN_free(ssl->priv);
 	ssl->priv = NULL;
+#endif
 
 	if (ssl->pkey)
 	    EVP_PKEY_free(ssl->pkey);
@@ -972,6 +978,7 @@ static void rpmsslVersionLog(void)
 }
 #endif
 
+#ifndef OPENSSL_NO_ECDSA
     {	size_t nc = EC_get_builtin_curves(NULL, 100);
 	EC_builtin_curve * c = alloca(nc * sizeof(*c));
 	size_t i;
@@ -986,6 +993,7 @@ static void rpmsslVersionLog(void)
 		o->sn, c[i].nid, pgpHexStr(o->data, o->length), o->ln);
 	}
     }
+#endif
 
     rpmlog(msglvl, "----------\n");
 }
@@ -1092,6 +1100,7 @@ assert(0);
       }	break;
     case PGPPUBKEYALGO_ECDSA:
       {
+#ifndef OPENSSL_NO_ECDSA
 	EC_KEY * ec = EVP_PKEY_get0(ssl->pkey);
 	ASN1_OBJECT * o = OBJ_nid2obj(ssl->nid);
 	unsigned char *q;
@@ -1110,6 +1119,7 @@ assert(0);
 	be += bn/8;
 assert(be == q);
 
+#endif
       }	break;
     }
 
@@ -1286,6 +1296,7 @@ assert(ssl->dsasig == NULL);
 	ssl->dsasig = NULL;
 	break;
     case PGPPUBKEYALGO_ECDSA:
+#ifndef OPENSSL_NO_ECDSA
 assert(ssl->ecdsasig == NULL);
 	q = ssl->sig;
 	ssl->ecdsasig = d2i_ECDSA_SIG(NULL, &q, ssl->siglen);
@@ -1306,6 +1317,7 @@ assert(ssl->ecdsasig == NULL);
 
 	ECDSA_SIG_free(ssl->ecdsasig);
 	ssl->ecdsasig = NULL;
+#endif
 	break;
     }
 
